openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/PatientResource.java

```diff
@@ -301,8 +299,7 @@ private static String normalizeName(String name) {
 
     protected static Patient jsonToPatient(SimpleObject json) {
         Patient patient = new Patient();
-        // TODO: do this properly from authentication
-        patient.setCreator(CREATOR);
+        patient.setCreator(Context.getAuthenticatedUser());
```

Comment:
This is actually necessary to ensure that data is consistent before and after sync. `CREATOR` was a fake `User`, that had been given ID `1` but wasn't otherwise populated. This logic relied upon the object being written to the DB and then being read from the DB again before being used.

That doesn't work with the sync module, because the sync module intercepts the record before it reaches the DB.